### PR TITLE
Handle cudf's leftsemi and leftanti joins

### PR DESCRIPTION
**Pull request by rjzamora**
Minor changes in to properly use `single_partition_join` with `how="leftsemi"` and `"leftanti"` (available in the `cudf` backend).

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
Thanks @rjzamora .  In principle I'm ok with this.  Are there tests somewhere for this, perhaps in cudf?

**Comment by rjzamora** [comment]
Thanks @mrocklin - I'll add tests here (and in cudf)

File: dask/dataframe/multi.py [hunk 1]
```diff
@@ -475,11 +480,12 @@ def merge(
         )
 
     # Single partition on one side
+    # Note that cudf supports "leftsemi" and "leftanti" joins
     elif (
         left.npartitions == 1
         and how in ("inner", "right")
         or right.npartitions == 1
-        and how in ("inner", "left")
+        and how in ("inner", "left", "leftsemi", "leftanti")
```
Comment (jrbourbeau) [inline dask/dataframe/multi.py]: Thanks @rjzamora! Small nitpick: could you create a variable which stores these values so we can reference it throughout the codebase (i.e. `how in allowed_hows`)

Comment (rjzamora) [reply dask/dataframe/multi.py]: Thanks for the review @jrbourbeau - Good suggestion.  Hopefully I addressed this.

**Comment by mrocklin** [comment]
Thanks @rjzamora !

**Pull request merged**